src/task/SubscribeAndReceiveSubscribeAckTask.ts

```diff
@@ -77,7 +80,25 @@ export default class SubscribeAndReceiveSubscribeAckTask extends BaseTask {
     const isSendingStreams: boolean =
       this.context.videoDuplexMode === SdkStreamServiceType.TX ||
       this.context.videoDuplexMode === SdkStreamServiceType.DUPLEX;
-    this.context.previousSdpOffer = new DefaultSDP(localSdp);
+
+    let compressedSDPOffer: Uint8Array | null;
+    const localSdpOffer = localSdp;
+
+    if (this.context.serverSupportsCompression) {
+      // If the server supports compression, then send the compressed version of the sdp
+      // and exclude the original sdp offer.
+      let prevOffer = '';
+      if (this.context.previousSdpOffer) {
+        prevOffer = this.context.previousSdpOffer.toString();
+      }
+      compressedSDPOffer = this.textCompressor.compress(localSdpOffer, prevOffer);
```

Comment:
nit: Can remove the above if statement with this.context.previousSdpOffer ? this.context.previousSdpOffer.toString() : ''

Reply:
this.context.previousSdpOffer can be null. so it will throw an error if we try to access the toString

Reply:
Yeah thats why we do a if check 
`this.context.previousSdpOffer ? this.context.previousSdpOffer.toString() : ''`